dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESMappingAPIImpl.java

```diff
@@ -286,29 +278,26 @@ public String toJson(final Contentlet contentlet) throws DotMappingException {
 			//add workflow to map
 			contentletMap.putAll(getWorkflowInfoForContentlet(contentlet));
 
-			if(UtilMethods.isSet(contentIdentifier.getSysPublishDate())) {
-				contentletMap.put(ESMappingConstants.PUBLISH_DATE, elasticSearchDateTimeFormat.format(contentIdentifier.getSysPublishDate()));
-				contentletMap.put(ESMappingConstants.PUBLISH_DATE + TEXT,
-						datetimeFormat.format(contentIdentifier.getSysPublishDate()));
-			}else {
-				contentletMap.put(ESMappingConstants.PUBLISH_DATE, elasticSearchDateTimeFormat.format(versionInfo.getVersionTs()));
-				contentletMap.put(ESMappingConstants.PUBLISH_DATE + TEXT,
-						datetimeFormat.format(versionInfo.getVersionTs()));
-			}
+            if (UtilMethods.isSet(contentIdentifier.getSysPublishDate())) {
+                contentletMap.put(ESMappingConstants.PUBLISH_DATE,
```

Comment:
this could be reduce to
```
contentletMap.put(ESMappingConstants.PUBLISH_DATE,
                        elasticSearchDateTimeFormat.format(UtilMethods.isSet(contentIdentifier.getSysPublishDate())?contentIdentifier.getSysPublishDate():versionInfo.getVersionTs()));
```
Up to you
```

Reply:
it could, but I think this way it's easier to read